gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/CopyJobInfo.java

```diff
@@ -0,0 +1,255 @@
+     */
+    public Builder sourceTables(List<TableId> sourceTables) {
+      this.sourceTables = sourceTables != null ? ImmutableList.copyOf(sourceTables) : null;
+      return self();
+    }
+
+    /**
+     * Sets the destination table of the copy job.
+     */
+    public Builder destinationTable(TableId destinationTable) {
+      this.destinationTable = destinationTable;
+      return self();
+    }
+
+    /**
+     * Sets whether the job is allowed to create new tables.
+     *
+     * @see <a href="https://cloud.google.com/bigquery/docs/reference/v2/jobs#configuration.link">
+     *     Jobs: Link Configuration</a>
+     */
+    public Builder createDisposition(CreateDisposition createDisposition) {
+      this.createDisposition = createDisposition;
+      return self();
+    }
+
+    /**
+     * Sets the action that should occur if the destination table already exists.
+     *
+     * @see <a href="https://cloud.google.com/bigquery/docs/reference/v2/jobs#configuration.link">
+     *     Jobs: Link Configuration</a>
+     */
+    public Builder writeDisposition(WriteDisposition writeDisposition) {
+      this.writeDisposition = writeDisposition;
+      return self();
+    }
+
+    @Override
+    public CopyJobInfo build() {
+      return new CopyJobInfo(this);
+    }
+  }
+
+  private CopyJobInfo(Builder builder) {
+    super(builder);
+    this.sourceTables = checkNotNull(builder.sourceTables);
+    this.destinationTable = checkNotNull(builder.destinationTable);
+    this.createDisposition = builder.createDisposition;
+    this.writeDisposition = builder.writeDisposition;
+  }
+
+  /**
+   * Returns the source tables to copy.
+   */
+  public List<TableId> sourceTables() {
+    return sourceTables;
+  }
+
+  /**
+   * Returns the destination table to load the data into.
+   */
+  public TableId destinationTable() {
+    return destinationTable;
+  }
+
+  /**
+   * Returns whether the job is allowed to create new tables.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/docs/reference/v2/jobs#configuration.copy">
+   *     Jobs: Copy Configuration</a>
+   */
+  public CreateDisposition createDisposition() {
+    return this.createDisposition;
+  }
+
+  /**
+   * Returns the action that should occur if the destination table already exists.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/docs/reference/v2/jobs#configuration.copy">
+   *     Jobs: Copy Configuration</a>
+   */
+  public WriteDisposition writeDisposition() {
+    return writeDisposition;
+  }
+
+  @Override
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  @Override
+  public MoreObjects.ToStringHelper toStringHelper() {
+    return super.toStringHelper()
+        .add("sourceTables", sourceTables)
+        .add("destinationTable", destinationTable)
+        .add("createDisposition", createDisposition)
+        .add("writeDisposition", writeDisposition);
+  }
+
+  @Override
+  public int hashCode() {
```

Comment:
Is this really necessary? equals already uses `toPb` which is redefined in all subclasses. Do you think we should redefine both equals and hashCode?